homeassistant/util/ssl.py

```diff
@@ -23,33 +78,56 @@ def create_no_verify_ssl_context() -> ssl.SSLContext:
         # This only works for OpenSSL >= 1.0.0
         sslcontext.options |= ssl.OP_NO_COMPRESSION
     sslcontext.set_default_verify_paths()
+    sslcontext.set_ciphers(SSL_CIPHER_LISTS[ssl_cipher_list])
+
     return sslcontext
 
 
-def client_context() -> ssl.SSLContext:
+def client_context(
+    ssl_cipher_list: SslCipherList = SslCipherList.MODERN,
+) -> ssl.SSLContext:
     """Return an SSL context for making requests."""
 
     # Reuse environment variable definition from requests, since it's already a
     # requirement. If the environment variable has no value, fall back to using
     # certs from certifi package.
     cafile = environ.get("REQUESTS_CA_BUNDLE", certifi.where())
 
-    return ssl.create_default_context(purpose=ssl.Purpose.SERVER_AUTH, cafile=cafile)
-
+    sslcontext = ssl.create_default_context(
+        purpose=ssl.Purpose.SERVER_AUTH, cafile=cafile
+    )
+    sslcontext.set_ciphers(SSL_CIPHER_LISTS[ssl_cipher_list])
 
-# Create this only once and reuse it
-_DEFAULT_SSL_CONTEXT = client_context()
-_DEFAULT_NO_VERIFY_SSL_CONTEXT = create_no_verify_ssl_context()
+    return sslcontext
 
 
-def get_default_context() -> ssl.SSLContext:
+# Create this only once and reuse it
+_DEFAULT_SSL_CONTEXT_MODERN = client_context(SslCipherList.MODERN)
+_DEFAULT_SSL_CONTEXT_INTERMEDIATE = client_context(SslCipherList.INTERMEDIATE)
+_DEFAULT_NO_VERIFY_SSL_CONTEXT_MODERN = create_no_verify_ssl_context(
+    SslCipherList.MODERN
+)
+_DEFAULT_NO_VERIFY_SSL_CONTEXT_INTERMEDIATE = create_no_verify_ssl_context(
+    SslCipherList.INTERMEDIATE
+)
+
+
+def get_default_context(
```

Comment:
I think it’s rare that an option is going to be passed here so we probably want to avoid creating the intermediate ones at startup. 

But since we only want to do it once we can wrap this in ‘@cache’ 

same for the function below

Reply:
To avoid caching the default one you could make it call another function if there is an argument and wrap that in `@cache` (from lru_tools)

Reply:
made `create_no_verify_ssl_context()` and `client_context()` cached, because they are used by other integrations (_`apache_kafka` , `imap_mail_content` , `smtp`_)

Reply:
I think we should keep this function as-is and not accept an argument.

Reply:
Sounds reasonable (_because they are called "get_default..."_).
Therefore we need to use the `client_context` and `create_no_verify_ssl_context` to get the correct ssl context